src/transxchange/read.rs

```diff
@@ -277,15 +283,139 @@ fn create_route(_transxchange: &Element, _vehicle_journey: &Element) -> Result<R
     })
 }
 
+struct NaiveDateRange {
+    current: NaiveDate,
+    end: NaiveDate,
+}
+
+impl NaiveDateRange {
+    fn new(start: NaiveDate, end: NaiveDate) -> NaiveDateRange {
+        if start <= end {
+            NaiveDateRange {
+                current: start,
+                end,
+            }
+        } else {
+            // If end date is smaller, then invert start and end
+            NaiveDateRange {
+                current: end,
+                end: start,
+            }
+        }
+    }
+}
+
+impl Iterator for NaiveDateRange {
+    type Item = NaiveDate;
+
+    fn next(&mut self) -> Option<Self::Item> {
```

Comment:
what's the need to have an iterator on only one `NaiveDateRange` ?
you can't do that in `new`?

Reply:
I'm not sure I understand the comment here... Do you say we do not need the `NaiveDateRange` structure and could inline all the code?

Reply:
why do you use an iterator for that?

Reply:
I'm iterating over a list of dates between a start and an end date. Just felt like an iterator was logical for iterating.

Reply:
ok